app/app/queue.py

```diff
@@ -11,19 +11,24 @@
 def add_to_queue():
     # Add a new entry to the queue
     body = request.get_json(force=True)
-
     try:
-        new = Queue(studentName=body['studentName'],
+        new_queue_entry = Queue(studentName=body['studentName'],
                     studentNumber=body['studentNumber'],
                     unitCode=body['unitCode'],
                     enquiry=body['enquiry'],
                     queue=body['queue'],
                     status='In Queue',
                     enterQueueTime=datetime.now())
-        db.session.add(new)
+        db.session.add(new_queue_entry)
+        db.session.commit()
+
+        # new_queue_entry only has the id after db commit
+        new_queue_event_log = QueueEventLog(new_queue_entry)
+        db.session.add(new_queue_event_log)
         db.session.commit()
-        print(new)  # This print is important (do not remove)
-        return new.to_dict(), 201
+
+        print(new_queue_entry)  # This print is important (do not remove) - Unknown reason (further investigation needed)
```

Comment:
Also very curious why these prints are needed

Reply:
This prints are necessary because if it doesn't exist, the endpoint doesn't return.

Why? Flask bug maybe?

Reply:
thats why it needs more investigation